### PR TITLE
Update to match akka persistence jvm 2.4.8

### PR DESCRIPTION
**Pull request by cconstantin**
* Avoid NRE in IsRecovering
* Fix InvalidActorNameException in event adapter specs
* Make sure ReplayFilter works OK if no previous writer ID was seen
* Remove Initialize from PersistentFSM API
* Recovery timeout for persistent actors

PersistentFSM is not yet up to date

### CODE REVIEW

**Comment by cconstantin** [comment]
Added https://github.com/akkadotnet/akka.net/issues/2246 for updating PersistentFSM

File: src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+        public PersistentActorRecoveryTimeoutSpec()
+            : base(SteppingMemoryJournal.Config(JournalId).WithFallback(
+                ConfigurationFactory.ParseString(
+                    @"akka.persistence.journal.stepping-inmem.recovery-event-timeout = 1000ms
+                    akka.actor.serialize-messages = off"))
+                  .WithFallback(Configuration("PersistentActoryRecoveryTimeoutSpec")))
+        {
+            Sys.EventStream.Publish(new Mute(new CustomEventFilter(e => e.Message is AskTimeoutException)));
+            //TODO: remove akka.actor.serialize-messages=off when Props serialization will be resolved (github issue: #569)
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs]: This issue has been closed a long time ago. 
https://github.com/akkadotnet/akka.net/issues/569

File: src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                _probe.Tell(new Status.Failure(reason));
+            }
+        }
+
+        #endregion
+
+        public PersistentActorRecoveryTimeoutSpec()
+            : base(SteppingMemoryJournal.Config(JournalId).WithFallback(
+                ConfigurationFactory.ParseString(
+                    @"akka.persistence.journal.stepping-inmem.recovery-event-timeout = 1000ms
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs]: Why 1000 ms, but not 100 ms?

File: src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                    akka.actor.serialize-messages = off"))
+                  .WithFallback(Configuration("PersistentActoryRecoveryTimeoutSpec")))
+        {
+            Sys.EventStream.Publish(new Mute(new CustomEventFilter(e => e.Message is AskTimeoutException)));
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs]: I can't find this line in scala version

File: src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            Watch(replaying);
+
+            // initial read highest
+            SteppingMemoryJournal.Step(journal);
+
+            probe.ExpectMsg<Status.Failure>().Cause.GetType().ShouldBe(typeof(RecoveryTimedOutException));
+            ExpectTerminated(replaying);
+
+            // avoid having it stuck in the next test from the last read request above
+            //SteppingMemoryJournal.Step(journal);
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs]: Do you know why it is happening?

**Comment by alexvaluyskiy** [comment]
We have not got EventAdapterSpec yet
https://github.com/akka/akka/blob/master/akka-persistence/src/test/scala/akka/persistence/EventAdapterSpec.scala

**Comment by cconstantin** [comment]
@alexvaluyskiy the EventAdapterSpec is named MemoryEventAdapterSpec in our source.

**Comment by cconstantin** [comment]
@alexvaluyskiy I've addressed your other comments regarding the recovery timeout spec. I had to initialize the journal before instantiating the persistent actor, otherwise the 100ms timeout would elapse before the journal started.

**Comment by Danthar** [comment]
Needs rebase

**Comment by cconstantin** [comment]
@Danthar rebased

**Comment by alexvaluyskiy** [comment]
proto file should be here `src/core/protobuf`

**Comment by alexvaluyskiy** [comment]
@cconstantin do you have short plans to update PersistentFSM?

**Comment by alexvaluyskiy** [comment]
have you implemented this PR?
https://github.com/akka/akka/pull/19850

**Comment by cconstantin** [comment]
@alexvaluyskiy I missed the protobuf folder, I removed that commit. No short term plans for PersistentFSM, I'm going to try and get the Cassandra plugin updated next. https://github.com/akka/akka/pull/19850 is already ported.

**Comment by alexvaluyskiy** [comment]
LGTM. But we have to wait for build agents. @Aaronontheweb

**Pull request merged**